src/locale/localed.c

```diff
@@ -670,13 +663,13 @@ static int method_set_x11_keyboard(sd_bus_message *m, void *userdata, sd_bus_err
                         sd_bus_message_get_bus(m),
                         "/org/freedesktop/locale1",
                         "org.freedesktop.locale1",
-                        "X11Layout", "X11Model", "X11Variant", "X11Options", NULL);
+                        "X11Layout", "X11Model", "X11Variant", "X11Options",
+                        convert ? "VConsoleKeymap" : NULL,
+                        convert ? "VConsoleKeymapToggle" : NULL,
+                        NULL);
 
-        if (convert) {
-                r = x11_convert_to_vconsole_and_emit(c, m);
-                if (r < 0)
-                        log_error_errno(r, "Failed to convert keymap data: %m");
-        }
+        if (convert)
+                (void) vconsole_reload(sd_bus_message_get_bus(m));
 
         return sd_bus_reply_method_return(m, NULL);
 }
```

Comment:
afaics this will make some file IO things fatal that previously weren't fatal, no?

I think it would make sense to make only issues with updating our internal idea what settings are in effect fatal. Any changes to the files on disk should probably be handled gracefully: logged about but otherwise ignored.

Reply:
rationale for the latter: we now write multiple files and we cannot possibly gurantee that either both updates worked or neither. unix fs apis simply have no concept that would allow us to implement that. hence I think we should update what we can, log about what we cannot, but return success as long as our in-memory state was updated.